docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+# Cross Chain EVM Swap
+In this section, we will demonstrate how you can use PKPs and Lit Actions to perform a cross-chain swap. There are three parts to this flow:
+
+- Understanding Lit PKPs as escrow
+- Lit Action
+- Implementing Mint/Grant/Burn with SDK
+
+## 1) Understanding Lit PKPs as escrow
+
+A Private key is meant for signing transactions which are executed on the blockchain, a network verifies if the transaction is a legitimate one and then executes it. With Lit Network we can generate a special type of private keys which can be programmed to sign transactions only when certain conditions are met. With this, we can create several escrow mechanisms, and we will look into how we can execute a cross-chain swap.
+
+### A swap example would be:
+Suppose Alice holds 100 token A on Base and Bob holds 100 token B on Chronicles and they both agree to a swap of,
+- 4 token A from Alice to Bob on Base
+- 8 token B from Bob to Alice on Chronicles
+
+We have an [example code](https://github.com/anshss/lit-evm-swap) for the below solution.
+
+### How does this work?
+When you generate a PKP on the LIT Network, a PKP NFT is first minted to your wallet, representing your ownership over the underlying private key residing on the LIT Network. This PKP NFT is responsible for managing auth methods over the private key.
+
+Lit Actions are programmable code written in Javascript which can run on the LIT Network. We can program a Lit Action so that if certain permissions are met for a pkp, it must sign specific transactions that can be sent to the blockchain. We need to grant a Lit Action as a permitted action to the PKP to allow the Lit Action to generate signatures with it.
+
+We can create an escrow pkp in the following way (Mint/ Grant/ Burn)
+
+- Mint a PKP on Lit
+- Grant a permitted Lit Action
+- Burn the PKP NFT so that no more auth methods can be modified
+
+This approach will allow us to create scenarios where a private key can generate signs based on conditions and unlock robust opportunities like cross-chain swapping, messaging or bridging.
+
+For a Swap Lit Action, both parties need to deposit their funds to the PKP and based on the conditions the PKP should be completing the swap:
+
+- Alice (or Bob) creates a Mint/Grant/Burn PKP
+- Alice deposits 4 token A on Base Sepolia to the PKP
+- Bob deposits 8 token B on Chronicles to the PKP
+- Alice (or Bob) executes the Lit Action.
+
+Lit Action execution, checks conditions and executes the swap from the PKP
+
+If the swap conditions aren't met then the Lit Action execution shall revert the funds to their depositors.
+
+### Architecture for Mint/Grant/Burn
+
+![accessControl](/img/mgb-flow.png)
+
+## 2) Lit Action
+
+Now we'll dive into the depths of the Lit Action and explore how we can conduct conditional signing.
+
+Below are the parameters that decide the swapping conditions.
```

Comment:
Can we provide more context on the example code? Something like "Below are the parameters for the cross-chain swap. In the example below, we'll be transferring 4 test ETH tokens on Sepolia for 8 (tstLPX?) tokens on Chronicle Yellowstone"